src/System.Memory/tests/Performance/Perf.ReadOnlySequence.Slice.cs

```diff
@@ -2,765 +2,366 @@
-            SequencePosition start = buffer.GetPosition(0);
-
-            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
-            {
-                int localInt = 0;
-                using (iteration.StartMeasurement())
-                {
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                    {
-                        ReadOnlySequence<byte> temp = buffer.Slice(start);
-                        localInt ^= temp.Start.GetInteger();
-                    }
-                }
-                _volatileInt = localInt;
-            }
+            public Segments010() : base(new ReadOnlySequenceFactory<byte>.SegmentsTestSequenceFactory(10)) { }
         }
 
-        [Benchmark(InnerIterationCount = InnerCount)]
-        [InlineData(10_000, 100)]
-        private static void Byte_Array_PosLong(int bufSize, int bufOffset)
+        public class Segments100 : Perf_ReadOnlySequence_Slice_Byte
         {
-            var buffer = new ReadOnlySequence<byte>(new byte[bufSize], bufOffset, bufSize - 2 * bufOffset);
-            long sliceLen = buffer.Length;
-            SequencePosition start = buffer.GetPosition(0);
-
-            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
-            {
-                int localInt = 0;
-                using (iteration.StartMeasurement())
-                {
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                    {
-                        ReadOnlySequence<byte> temp = buffer.Slice(start, sliceLen);
-                        localInt ^= temp.Start.GetInteger();
-                    }
-                }
-                _volatileInt = localInt;
-            }
+            public Segments100() : base(new ReadOnlySequenceFactory<byte>.SegmentsTestSequenceFactory(100)) { }
         }
+    }
 
-        [Benchmark(InnerIterationCount = InnerCount)]
-        [InlineData(10_000, 100)]
-        private static void Byte_Array_PosPos(int bufSize, int bufOffset)
-        {
-            var buffer = new ReadOnlySequence<byte>(new byte[bufSize], bufOffset, bufSize - 2 * bufOffset);
-            SequencePosition start = buffer.GetPosition(0);
-            SequencePosition end = buffer.GetPosition(0, buffer.End);
-
-            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
-            {
-                int localInt = 0;
-                using (iteration.StartMeasurement())
-                {
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                    {
-                        ReadOnlySequence<byte> temp = buffer.Slice(start, end);
-                        localInt ^= temp.Start.GetInteger();
-                    }
-                }
-                _volatileInt = localInt;
-            }
-        }
-
-        #endregion
-
-        #region Byte_Memory
+    public abstract class Perf_ReadOnlySequence_Slice_Char: Perf_ReadOnlySequence_Slice<char>
```

Comment:
Why is this class (and the byte class) abstract?

Reply:
Because they should not be created by xUnit library for testing. Only children creates correct sequence generator and can be used by xUnit. For example why `public abstract class ReadOnlySequenceTestsByte` is abstract https://github.com/dotnet/corefx/blob/master/src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.byte.cs ?